app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/service.ts

```diff
@@ -0,0 +1,897 @@
+            asset => ({ ...asset, type: asset.id.match(/^(.+?)-/)?.[1] } as Asset)
+        )
+    }
+
+    /** Creates a directory, on the Cloud backend API.
+     *
+     * @throws An error if a 401 or 404 status code was received. */
+    async createDirectory(body: CreateDirectoryRequestBody): Promise<Directory> {
+        const response = await this.post<Directory>(CREATE_DIRECTORY_PATH, body)
+
+        if (
+            response.status === http.HttpStatus.unauthorized ||
+            response.status === http.HttpStatus.notFound
+        ) {
+            throw new Error('Unable to create directory.')
+        }
+
+        return await response.json()
+    }
+
+    /** Returns a list of projects belonging to the current user, from the Cloud backend API.
+     *
+     * @throws An error if status code 401 or 404 was received.
+     */
+    async listProjects(): Promise<ListedProject[]> {
+        const response = await this.get<ListProjectsResponseBody>(LIST_PROJECTS_PATH)
+
+        if (
+            response.status === http.HttpStatus.unauthorized ||
+            response.status === http.HttpStatus.notFound
+        ) {
+            throw new Error('Unable to list projects.')
+        }
+
+        return (await response.json()).projects
+    }
+
+    /** Creates a project for the current user, on the Cloud backend API.
+     *
+     * @throws An error if a 401 or 404 status code was received. */
+    async createProject(body: CreateProjectRequestBody): Promise<CreatedProject> {
+        const response = await this.post<CreatedProject>(CREATE_PROJECT_PATH, body)
+
+        if (
+            response.status === http.HttpStatus.unauthorized ||
+            response.status === http.HttpStatus.notFound
+        ) {
+            throw new Error('Unable to create project.')
+        }
+
+        return await response.json()
+    }
+
+    /** Closes the project identified by the given project ID, on the Cloud backend API.
+     *
+     * @throws An error if a 401 or 404 status code was received. */
+    async closeProject(projectId: ProjectId): Promise<void> {
+        const response = await this.post(closeProjectPath(projectId), {})
+
+        if (
+            response.status === http.HttpStatus.unauthorized ||
+            response.status === http.HttpStatus.notFound
+        ) {
+            throw new Error('Unable to close project.')
+        }
+    }
+
+    /** Returns project details for the specified project ID, from the Cloud backend API.
+     *
+     * @throws An error if a 401 or 404 status code was received. */
+    async getProjectDetails(projectId: ProjectId): Promise<Project> {
+        const response = await this.get<Project>(getProjectDetailsPath(projectId))
+
+        if (
+            response.status === http.HttpStatus.unauthorized ||
+            response.status === http.HttpStatus.notFound
+        ) {
+            throw new Error('Unable to get project details.')
+        }
```

Comment:
According to our styling docs, in such cases, we prefer to use one-line well named assignments, like:

```
let goodName = response.status === http.HttpStatus.unauthorized
let anotherGoodName = response.status === http.HttpStatus.notFound
if (goodName || anotherGoodName) { ... }
```

Can we reformat it this way, please? :)

Reply:
Also, a minor thing, but, why do we use so much newlines in these functions? If you feel that the body should be split into parts, maybe we should refactor them to other functions then? Not using so much newlines makes the code shorter, and easier to read, as you can visually see connected code chunks faster.